### PR TITLE
nn models rewrite

### PR DESCRIPTION
**Pull request by krsnapaudel**
Tried to simplify nn-models code.
1. Skip early stopping and saving best model.
2. Hyperparameter optimization and confusion about what is in `settings`, `args` and `kwargs`.
3. `predict_batch` and `predict`.
4. Transforms code. Separating time series and static inputs must be done even when transforms is empty.
5. Add validation of AgML Workshop 2024 results.

### CODE REVIEW

File: cybench/runs/run_benchmark.py [hunk 1]
```diff
@@ -40,41 +40,23 @@
...
+    "select_features": True,
+    "optimize_hyperparameters": True,
+}
+
+_BASELINE_MODEL_FIT_KWARGS["SklearnRF"] = {
     "optimize_hyperparameters": True,
-    "param_space": {"estimator__alpha": [0.01, 0.1, 0.0, 1.0, 5.0, 10.0]},
 }
 
 _BASELINE_MODEL_FIT_KWARGS["LSTM"] = {
```
Comment (aikepotze) [inline cybench/runs/run_benchmark.py:61]: Looks like device is not specified in current LSTM fit kwargs, defaulting it to CPU. Should be added to default fit kwargs.

**Review by aikepotze — CHANGES_REQUESTED** [review]
Rewritten nn methods look good to me, simpler than the previous implementation. Only issue I could find is the missing device call in the run_benchmark script, leading to training on CPU.

Aside from that, maybe we want to extend the hyperparameter tuning to include initialization arguments as well (as this is a common use case). However, this would need changes to the class structure, as the base nn model cannot re-initialize itself. What do you think?

File: cybench/runs/run_benchmark.py (hunk 1 above)
Comment (krsnapaudel) [reply cybench/runs/run_benchmark.py:51]: It's addressed in latest changes.

**Comment by krsnapaudel** [comment]
@aikepotze: "However, this would need changes to the class structure, as the base nn model cannot re-initialize itself. What do you think?" I don't the model needs to reinitialize the model. We have saved `init_args` for models we create to optimize hyperparameters. That I think is sufficient.

**Comment by krsnapaudel** [comment]
I have validated results from AgML workshop. Now the workshop implementation of `LSTMModel` and `ExampleLSTM` included in the benchmark have similar results. The architectures are similar. Some differences:
1. `LSTMModel` relies on batch norm for normalizing inputs and intermediate features. `ExampleLSTM` normalizes inputs on its own. There is no batch norm after LSTM layer.
2. `LSTMModel` does not use `LRScheduler`; `ExampleLSTM` does.

The NRMSE on test years [2012, 2018] is approximately 25% for both models.
